replica/table.cc

```diff
@@ -734,6 +735,25 @@ table::try_flush_memtable_to_sstable(lw_shared_ptr<memtable> old, sstable_write_
     co_return co_await with_scheduling_group(_config.memtable_scheduling_group, std::ref(try_flush));
 }
 
+future<> table::maybe_wait_for_sstable_count_reduction() {
+    while (true) {
+        if (_async_gate.is_closed()) {
+            co_return;
+        }
+        auto sstable_runs_with_memtable_origin = boost::copy_range<std::unordered_set<utils::UUID>>(
+            *_main_sstables->all()
+            | boost::adaptors::filtered([] (const sstables::shared_sstable& sst) {
+                return sst->get_origin() == "memtable";
+            })
+            | boost::adaptors::transformed(std::mem_fn(&sstables::sstable::run_identifier)));
+        if (sstable_runs_with_memtable_origin.size() < schema()->max_compaction_threshold() || is_auto_compaction_disabled_by_user()) {
```

Comment:
we can probably forgive # of runs == 32. so <= instead.

also let's make the threshold a maximum of schema()->max_compaction_threshold() and 32, so making it safe for users who mistakenly decreased max threshold to a low value.